gluten-core/src/main/scala/org/apache/gluten/extension/columnar/OffloadSingleNode.scala

```diff
@@ -232,6 +233,38 @@ case class OffloadFilter() extends OffloadSingleNode with LogLevelUtil {
   }
 }
 
+case class OffloadWindow() extends OffloadSingleNode with LogLevelUtil {
+  import OffloadOthers._
+  private val replace = new ReplaceSingleNode()
+
+  override def offload(plan: SparkPlan): SparkPlan = plan match {
+    case window: WindowExec =>
+      if (TransformHints.isNotTransformable(window)) {
+        return window
+      }
+
+      val transformer = replace.doReplace(window)
+      val newChild = transformer.children.head match {
+        case SortExec(_, false, child, _)
+            if outputOrderSatisfied(child, transformer.requiredChildOrdering) =>
+          child
+        case p @ ProjectExec(_, SortExec(_, false, child, _))
+            if outputOrderSatisfied(child, transformer.requiredChildOrdering) =>
+          p.copy(child = child)
+        case children => children
+      }
+      transformer.withNewChildren(Seq(newChild))
+    case other => other
+  }
+
+  private def outputOrderSatisfied(child: SparkPlan, required: Seq[Seq[SortOrder]]): Boolean = {
+    Seq(child).zip(required).forall {
+      case (child, requiredOrdering) =>
+        SortOrder.orderingSatisfies(child.outputOrdering, requiredOrdering)
+    }
+  }
```

Comment:
Could just assume an unary node is returned from `replace.doReplace(window)` then we can avoid `zip` to simplify code.